### DISCUSSION TITLE
Cancel Request for failed SDK Call / API Calls Timing Out (Gateway Timeout)

### DISCUSSION THREAD

**Discussion by bllchmbrs** [deepgram/community#320] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

Right now I have a request that is pending in the API. I sent the same request twice, but then my code failed, and so the process exited (leaving the process).

I now have 2 stuck, pending, requests that are never going to complete (and 1 seemed to cause the other).

How can I delete / cancel these requests? I couldn't find anything in the documentation and had no problems with other transcriptions.

<img width="670" alt="image" src="https://github.com/deepgram/community/assets/1642503/7d5b9f2b-1a66-495b-81fb-a558b53cec85">

**Comment by jkroll-deepgram** [comment]
Hi @bllchmbrs, all requests that Deepgram receives will eventually either complete or fail. Now that more time has passed, do you see that their status has changed? As you found through lack of evidence in our docs, there is no ability or need to delete or cancel a request once it has been submitted.

If you share a specific request ID that you are interested in, I can also look into how the request was processed and resolved on Deepgram's side.

**Reply by bllchmbrs** [reply]
They are now "unknown". So we're good now. If they don't "block" other transcriptions from taking place then I don't see too much of an issue, but I worry that if some fail and i have no way of cancelling that i can't continue transcribing other audio. If that doesn't happen, even for the same file, then we're all good!

**Reply by jkroll-deepgram** [reply]
Rest assured that any pending requests will not prevent you from continuing to submit requests and receive transcripts back!

**Reply by bllchmbrs** [reply]
@jkroll-deepgram , I don't think I figured  this out actually. For some reason, I have some particular audio files that just don't transcribe. How can I debug what is wrong with the files?

It doesn't matter if I upload them via URL or if I do them via binary. They just don't convert and hang for 40+ minutes (it's maybe 1.5 hours of content).

**Reply by jkroll-deepgram** [reply, marked as answer]
[edited to correct accuracy regarding audio and response times]

@bllchmbrs Looking at your requests for the past week, they are all succeeding, generally within a minute. As you've looked into this more, are you getting back the responses you're expecting?

You can also look into our [Callback](https://developers.deepgram.com/docs/callback) feature, where you provide a URL that Deepgram will return results to asynchronously when the transcript is ready, rather than waiting synchronously for an hour for a response. 

----

If needed, depending on the nature of your audio and if your use case is latency-sensitive, is you can segment out an hours-long audio file into multiple shorter files, and submit them in parallel for a faster absolute response. However, this does have downsides of managing multiple transcripts for a single file, and Deepgram losing context across the segments, so I'd only recommend this if it would be a major help to trim down the latency from 60 to 15 minutes, for instance, when splitting a file into 4 quarters.

If you happen to be submitting video files instead of audio files, I strongly recommend pre-processing the media and submitting audio-only to Deepgram, as that by itself will greatly decrease latency.

**Reply by bllchmbrs** [reply]
Thanks!

**Reply by bllchmbrs** [reply]
@jkroll-deepgram , just to follow up on this for clarity.

I have 2 options enabled, entity detection and topic detection. These were significantly raising the time to execute AND they caused failures - I hit gateway timeouts. They were audio files over 2 hours.

One 'opportunity' would be to make it clearer that this will (a) significantly increase execution time (the same files are done much faster without those options) and (b) cost - I didn't know that they were costing me extra.

Hope this helps others too.

**Reply by jkroll-deepgram** [reply]
@bllchmbrs, thanks for sharing this follow-up info. We are working on improving our pricing page for clarity. For now, you can see what we call "Audio Intelligence add-ons" on the [pricing page](https://deepgram.com/pricing), which includes summarization, entity detection, PII redaction, and topic detection. These features are more expensive because they are more compute-intensive, which also makes them slower. Sorry to hear that they were also causing more failures - I will pass that feedback on to the team.
